src/semant/findescape.sml

```diff
@@ -0,0 +1,82 @@
+structure FindEscape: sig val findEscape: Absyn.exp -> unit
+                      end =
+struct
+  type depth = int
+  type escEnv = (depth * bool ref) Symbol.table
+
+  fun traverseVar(env:escEnv, d:depth, s:Absyn.var) : unit =
+    (case s of
+        Absyn.SimpleVar(sym,pos) =>
+          (case Symbol.look(env,sym) of
+              SOME(d',esc) => if d > d' then esc := true else ()
```

Comment:
@soelgary to be sure I understand. This is saying that the variables defined in a scope above our current scope escape, and therefor need to be linked.